openupgrade_scripts/scripts/mail/16.0.1.10/upgrade_analysis_work.txt

```diff
@@ -0,0 +1,222 @@
+mail         / fetchmail.server         / display_name (char)           : previously in module fetchmail
+mail         / fetchmail.server         / is_ssl (boolean)              : previously in module fetchmail
+mail         / fetchmail.server         / message_ids (one2many)        : previously in module fetchmail
+mail         / fetchmail.server         / name (char)                   : previously in module fetchmail
+mail         / fetchmail.server         / object_id (many2one)          : previously in module fetchmail
+mail         / fetchmail.server         / original (boolean)            : previously in module fetchmail
+mail         / fetchmail.server         / password (char)               : previously in module fetchmail
+mail         / fetchmail.server         / port (integer)                : previously in module fetchmail
+mail         / fetchmail.server         / priority (integer)            : previously in module fetchmail
+mail         / fetchmail.server         / script (char)                 : previously in module fetchmail
+mail         / fetchmail.server         / server (char)                 : previously in module fetchmail
+mail         / fetchmail.server         / server_type (selection)       : previously in module fetchmail
+mail         / fetchmail.server         / state (selection)             : previously in module fetchmail
+mail         / fetchmail.server         / user (char)                   : previously in module fetchmail
+# NOTHING TO DO: modules merged in apriori.py
+
+mail         / ir.actions.server        / mail_post_autofollow (boolean): NEW hasdefault: compute
+mail         / ir.actions.server        / mail_post_method (selection)  : NEW selection_keys: ['comment', 'email', 'note'], hasdefault: compute
+# NOTHING TO DO: New fields
+mail         / ir.actions.server        / state (False)                 : selection_keys is now '['code', 'followers', 'mail_post', 'multi', 'next_activity', 'object_create', 'object_write']' ('['code', 'email', 'followers', 'multi', 'next_activity', 'object_create', 'object_write']')
+# DONE (pre-migration): selection_key 'email' has been renamed 'mail_post'
+
+mail         / ir.mail_server           / mail_template_ids (one2many)  : NEW relation: mail.template
+# NOTHING TO DO: new field
+
+mail         / mail.channel             / _inherits                     : DEL _inherits: {'mail.alias': 'alias_id'}
+mail         / mail.channel             / alias_id (many2one)           : DEL relation: mail.alias, required
+# NOTHING TO DO: removed mail.alias.mixin from channel. See commit https://github.com/odoo/odoo/commit/ac00d19c8db572932a3863ab21a9a1f432dd93b5
+mail         / mail.channel             / channel_last_seen_partner_ids (one2many): DEL relation: mail.channel.partner
+mail         / mail.channel             / channel_member_ids (one2many) : NEW relation: mail.channel.member
+# DONE (pre-migration): field was renamed when calling rename_models
```

Comment:
Does the value of the column `channel_last_seen_partner_ids` be copied to `channel_member_ids` ?

I'm not sure that renaming the model `mail.channel.partner` to `mail.channel.member` would be enough to keep values from the column `channel_last_seen_partner_ids` to `channel_member_ids`. Is it enough ?

Reply:
Good question ! (and I started to write the use case "rename_model" yesterday...)

AFAIK, table should be renamed. 
https://github.com/OCA/openupgradelib/blob/f87f553aca1e8fb8447bc152473518c6b070ed1a/openupgradelib/openupgrade.py#LL872C1-L874C45

I'm not sure, thought how it has to be done.
(pseudo code)
A) rename_fields( channel_last_seen_partner_ids -> channel_member_ids)
B) rename_columns( channel_last_seen_partner_ids -> channel_member_ids)
C) rename_tables(channel_last_seen_partner_rel -> channel_member_rel)

@MiquelRForgeFlow, @pedrobaeza : any idea, as a regular contributor of OU scripts ?

Reply:
A) yes, that's needed.
B) no, it's one2many.
C) no, it's not many2many.

Reply:
Thanks a lot for your reactivity !

Reply:
Thanks for your replies. :) So how do we proceed. Should we merge this PR and then make a fix to add the rename of this field, or should I make a new PR with the work of @marielejeune with a commit fixing the rename of this field ?

Reply:
Hi, sorry I have a lot of work for the moment. My plan is to work on it during the Spanish OCA days next week :-)

Reply:
Hi @MiquelRForgeFlow, I'm not sure why the field must be renamed? 
I made a test migrating from 15 to 16 and the `mail.channel.partner` records of `channel_last_seen_partner_ids` in v15 are still preserved (as `mail.channel.member` records) in `channel_member_ids` field. 
I'm in Valencia if you want to discuss about it.

Reply:
Hi @marielejeune 
One thing is the data in the mail_channel_member, which as you say it's preserved, and another thing is the ir_model_fields data. By doing the field rename of channel_last_seen_partner_ids to channel_member_ids (using rename_fields), you are preserving the ir.model.fields data instead of forcing new one.

Reply:
Done, could you check?

Reply:
Good!